src/pal/inc/unixasmmacrosarm64.inc

```diff
@@ -0,0 +1,270 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+.macro PATCH_LABEL Name
+        .global C_FUNC(\Name)
+C_FUNC(\Name):
+.endm
+
+.macro LEAF_ENTRY Name, Section
+        .global C_FUNC(\Name)
+        .type \Name, %function
+C_FUNC(\Name):
+        .cfi_startproc
+.endm
+
+.macro LEAF_END_MARKED Name, Section
+        .global C_FUNC(\Name\()_End)
+C_FUNC(\Name\()_End):
+        .size \Name, .-\Name
+        .cfi_endproc
+.endm
+
+.macro PREPARE_EXTERNAL_VAR Name, HelperReg
+        ldr \HelperReg, [pc, #C_FUNC(\Name)@GOTPCREL]
+.endm
+
+.macro PROLOG_STACK_ALLOC Size
+        sub sp, sp, \Size
+        .cfi_adjust_cfa_offset \Size
+.endm
+
+.macro EPILOG_STACK_FREE Size
+        add sp, sp, \Size
+.endm
+
+.macro EPILOG_STACK_RESTORE
+        mov sp, fp
+.endm
+
+.macro PROLOG_SAVE_REG reg, ofs
+        str \reg, [sp, \ofs]
+.endm
+
+.macro PROLOG_SAVE_REG_PAIR reg1, reg2, ofs
+        stp \reg1, \reg2, [sp, \ofs]
+        .ifc \reg1, fp
+        mov fp, sp
+        .endif
+.endm
+
+.macro PROLOG_SAVE_REG_PAIR_INDEXED reg1, reg2, ofs
+        stp \reg1, \reg2, [sp, \ofs]!
+        .ifc \reg1, fp
+        mov fp, sp
+        .endif
+.endm
+
+.macro EPILOG_RESTORE_REG reg, ofs
+        ldr \reg, [sp, \ofs]
+.endm
+
+.macro EPILOG_RESTORE_REG_PAIR reg1, reg2, ofs
+        ldp \reg1, \reg2, [sp, \ofs]
+.endm
+
+.macro EPILOG_RESTORE_REG_PAIR_INDEXED reg1, reg2, ofs
+        ldp \reg1, \reg2, [sp], \ofs
+.endm
+
+.macro EPILOG_RETURN
+        ret
+.endm
+
+.macro EMIT_BREAKPOINT
+        brk #0
+.endm
+
+//-----------------------------------------------------------------------------
+// Define the prolog for a TransitionFrame-based method. This macro should be called first in the method and
+// comprises the entire prolog (i.e. don't modify SP after calling this).The locals must be 8 byte aligned 
+//
+.macro PROLOG_WITH_TRANSITION_BLOCK extraLocals, SaveFPArgs
```

Comment:
Could you please add the stack layout description like we have for AMD64 and ARM? It is very helpful.